tests/StatsdServer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef STATSD_SERVER_HPP
+#define STATSD_SERVER_HPP
+
+// TODO: should we actually add this to the distributed headers? With a tiny bit of work
+//  someone could write their own c++ statsd server
+
+
+class StatsdServer {
```

Comment:
so originally i started testing by making statsd a submodule to the repo and was going to actually run it with node etc but there were a couple things i didnt like about it:

- i have to install node just to test it, this takes a bunch of time in CI and can go down from time to time
- the testing methodology is messy. 
  - you have to start up statsd and possibly another backend in the background and then kill them at some point
  - you have to collect the results either in a log file or in a database and then pull that out and transcode it to something you can easily validate
  - validation has to be done with yet another script

so instead i opted to do a much cleaner test all in process. the statsd format is so simple it didnt make sense to do all of this other complicated stuff when i can just write a tiny bit of code to suck bytes out of a UDP socket and compare them so thats what i did. this also has the benefit of allowing downstream users to test that their own application works as they expect in terms of reporting metrics.